torch/utils/debug_mode.py

```diff
@@ -0,0 +1,134 @@
+import torch
+from torch.utils._python_dispatch import TorchDispatchMode
+import torch.utils._pytree as pytree
+
+from collections.abc import Callable
+from typing import Union, List, Optional, Iterable
+from functools import partial
+
+
+def embedding_check(func, args, kwargs):
+    invalid_index_mask = args[0].shape[0] <= args[1]
+    if torch.any(invalid_index_mask):
+        msg = (f"{func}: Received invalid indices for embedding matrix of shape : {args[0].shape}."
+               f" Invalid indices are {torch.masked_select(args[1], invalid_index_mask)}")
+        raise IndexError(msg)
+
+def extermal_check(func, args, kwargs, result, check):
+
+    def check_fn(check, check_method):
+        def error_if(t):
+            if check_method(t):
+                raise ValueError(f"{func} produced `{check}` in output")
+
+        pytree.tree_map_only(torch.Tensor, lambda t: error_if(t), result)
+
+    assert check in ('inf', 'nan')
+
+    if check == 'inf':
+        check_fn(check, lambda t: t.isinf().all())
+    elif check == 'nan':
+        check_fn(check, lambda t: t.isnan().all())
+
+
+def indexing_checks(func, args, kwargs):
+    if func in (torch.ops.aten.embedding, torch.ops.aten._embedding_bag):
+        embedding_check(func, args, kwargs)
+
+nan_check = partial(extermal_check, check='nan')
+inf_check = partial(extermal_check, check='nan')
+
+PRE_CHECKS = {'index': indexing_checks}
+POST_CHECKS = {'nan': nan_check, 'inf': inf_check}
+CHECKS = list(PRE_CHECKS.keys()) + list(POST_CHECKS.keys())
+
+def get_pre_checks(check):
+    # TODO: Complete this function
+    if check == 'all':
+        return tuple(PRE_CHECKS.values())
+    if isinstance(check, str):
+        return (PRE_CHECKS[check],)
+
+    result = []
+    for c in check:
+        if c in PRE_CHECKS:
+            result.append(PRE_CHECKS[c])
+
+    return tuple(result)
+
+
+def get_post_checks(check):
+    # TODO: Complete this function
```

Comment:
What do you mean by "Complete this function"? Is there more to do here?

Reply:
I think, I wanted to make sure that covers all cases. Will check that and remove the TODO.

Reply:
Done.